File: quick_union.py
def ID_Init(id_length):
    c=0
    id=[]
    while c<id_length:
        
        id.append(c)
        c+=1
    return id


def CheckIfRoot(id,i):
    while i != id[i]:
        i=id[i]
    return i        

def QuickUnion(id,p,q):
    i=CheckIfRoot(id,p)
    j=CheckIfRoot(id,q)
    id[i]=j
    return id;

File: quick_union.js
function CheckIfRoot(id,i)
{
    while(i!=id[i])
    {i=id[i];}
    return i;
}

function QuickUnion(id,p,q)
{
 var i=CheckIfRoot(id,p);   
 var j=CheckIfRoot(id,q);
 id[i]=j;
 return id;
}